src/Agent/NewRelic/Agent/Core/Logging/Logger.cs

```diff
@@ -230,11 +216,7 @@ public void Finest(Exception exception)
         /// </summary>
         public void FinestFormat(string format, params object[] args)
         {
-            if (IsFinestEnabled)
```

Comment:
I suspect these checks were deliberate, in order to avoid the overhead of formatting the string if we aren't going to log the line. I don't know how much of a performance hit it is, but I could see it being noticeable at Finest.

Reply:
Yes, they were added a while ago as part of a performance improvement.

Reply:
We have many other places where the strings are formatted directly in `Log.Debug()` or `Log.Finest()`.

`Log.FinestFormat()` is called in about 10 places and `Log.Finest()` with a string format is called in about 12 places.

`Log.DebugFormat()` is called in about 40 places, and `Log.Debug()` with a direct format string is called a similar number of times. 

So if performance is *really* a concern and if checking whether the log is enabled makes a noticeable difference, we ought to go back through the code and  clean up all the stray usages of `Log.Finest()` and `Log.Debug()`.